tools/update_vendor.sh

```diff
@@ -17,5 +19,11 @@ find vendor -type f \( \
     -o -name BUILD.bazel \
     -o -name '*.bzl' \
   \) -delete
+
+# create a symlink in vendor directory pointing cloud-provider-gcp/providers to the //providers.
+# This lets other packages and tools use the local staging components as if they were vendored.
+rm -fr "${KUBE_ROOT}/vendor/k8s.io/cloud-provider-gcp/providers"
+ln -s "../../../providers" "${KUBE_ROOT}/vendor/k8s.io/cloud-provider-gcp/providers"
```

Comment:
Would be good to not use relative paths. Lets base this off of KUBE_ROOT for the source as well. Easier to understand and that way as long as KUBE_ROOT is correct, we can be run from the wrong place, change directory in an earlier step, etc. and everything still works.

Reply:
Since the symlinks are checked in, we don't want an expanded environment variable (to wherever the clone of the repo is on a developer's box) to be used. I think this is why k8s/k8s also uses relative paths.

Reply:
Agree with Joe. FYI: https://github.com/kubernetes/cloud-provider-gcp/blob/8b9e1c69118cedef50ece703a01c4c5a26dac1df/vendor/k8s.io/cloud-provider-gcp/providers#L1

Reply:
Fair. Might be nice to ensure we're in the directory we expect, otherwise this will do something odd.